cfpb/cfgov-refresh
# Careers: vacancy announcement updates

Updates to job listing page

## Additions

- Update JobListing page layout:
  - Add `Who may apply` to definition list at top of job listing page
  - Separate `Description` section into `Summary`, `Responsibilities`, `Travel Required`, and optional `Additional content` fields
  - Make description section expandable on mobile
  - Add job application video 
  - Update sidebar content
- Simplify Applicant Type section of vacancy announcement page:
  - Break job type information out into `Job Length` and `Service Type` models with corresponding fields on JobListingPage
  - Show `Position type` as part of job description section
  - Add a `Display title` field to Applicant Type model to allow for more compact titles
   - Rename section to `Apply for this Position` and change button text
- Add `JobPosting` metadata for inclusion in Google's highlighted job search results and ping Google when job posting pages are published

## Changes
- Move `read-more.js` into its own file and require it in Ask CFPB answer and JobListing pages

## Testing

1. Check out branch and run migrations
2. Run `python cfgov/manage.py runscript create_career_types`
3. In Wagtail admin, update the description section on an existing JobListing page and verify that description, job type, applicant type(s), and other content display correctly on desktop and mobile
4. View source of JobListing page and verify that `JobPosting` metadata appears at bottom of page (it can be tested with [Google's structured data tool here](https://search.google.com/structured-data/testing-tool))
5. Set sitemap environment variables, publish a JobListing page, and verify that Google is notified of sitemap update
6. Verify expandable description content on mobile
 

## Screenshots

### Job Posting page
| Desktop | Mobile |
| --- | --- |
| ![desktop-pr](https://user-images.githubusercontent.com/778171/36082532-3c2c222c-0f5f-11e8-987c-d6f2c2462c04.png) | ![mobile-pr](https://user-images.githubusercontent.com/778171/36082533-426d003e-0f5f-11e8-96e8-d452ca4ee1a2.png) |

### Changes to description section in Wagtail

![description](https://user-images.githubusercontent.com/778171/36082765-3bb43f58-0f61-11e8-904d-f7c10994e230.png)


### New Job Length and Service Type models

<img width="484" alt="screen shot 2018-02-11 at 7 00 07 pm" src="https://user-images.githubusercontent.com/778171/36082770-4ce35fde-0f61-11e8-91e4-0caca6839dbc.png">
<img width="616" alt="screen shot 2018-02-11 at 6 59 58 pm" src="https://user-images.githubusercontent.com/778171/36082771-4d0c0e7a-0f61-11e8-8110-bd59e4a67f07.png">
<img width="848" alt="screen shot 2018-02-11 at 7 00 36 pm" src="https://user-images.githubusercontent.com/778171/36082783-62823356-0f61-11e8-8026-b01cac892317.png">






## Todos

- Sidebar content is still being finalized

## Checklist
